src/helpers/circular_map.rs

```diff
@@ -93,17 +87,26 @@ impl<K: Clone + PartialEq, V: Clone, const N: u32> CircularMap<K, V, N> {
     /// Removes the key-value pair for the given key from the circular map.
     ///
     pub fn remove(&mut self, key: &K) {
-        let mut queue = CircularQueue::with_capacity(N as usize);
-        for (k, v) in self.queue.iter() {
-            if key != k {
-                queue.push((k.clone(), v.clone()));
+        if self.map.remove(key).is_some() {
+            for k in self.queue.asc_iter_mut() {
+                if k.as_ref() == Some(key) {
+                    *k = None;
+                    return;
+                }
             }
         }
-        self.queue = queue;
+    }
+
+    ///
+    /// Removes all the entries from the circular map.
+    ///
+    pub fn clear(&mut self) {
```

Comment:
using this method instead of replacing the whole map with `Default::default` reuses the existing allocations instead of re-allocating